app.ts

```diff
@@ -0,0 +1,171 @@
+            }
+            $("#ContentBody").empty();
+            $("#ContentBody").append(appendable);
+        });
+    return [fromID as number, toID as number];
+}
+
+// Credit: https://gist.github.com/ca0v/73a31f57b397606c9813472f7493a940
+
+function placeRecordsFromCursor(cursor: Array<number>) {
+    cursor = cursor.sort((a,b) => {return a-b});
+    return placeRecords(cursor[0], cursor[1])
+}
+//#endregion
+
+//#region Handlers
+function getPageContent(fromID: number, toID: number) {
+    $("#HeaderRow").empty();
+    getColumnNames()
+        .then((columns: Array<string>) => {
+            for (const column of columns) {
+                let writable: string = '<th align="center">' + column + '</th>';
+                $("#HeaderRow").append(writable);
+            }
+        });
+    
+    return placeRecords(fromID, toID);
+}
+
+function toNumber(input: string | number, output: any = 1) {
+    if (typeof input === "string") {
+        if (output == 1) {
+            return parseInt(input);
+        } else {
+            return parseFloat(input);
+        }
+    }
+    else if (typeof input === "number") {
+        return input;
+    }
+    else {
+        return 0;
+    }
+}
+function nextPageResize(previousCursor: Array<number>): number {
+        let fromID = toNumber(previousCursor.sort((a, b) => {return a - b})[0]);
+        let toID = toNumber(previousCursor.sort((a, b) => {return a - b})[1]);
+
+        const documentHeight = ($(window).innerHeight() as number) - ($("#" + fromID.toString()).height() as number);
+        for (let i = fromID; i <= toID; i++) {
+            let element = ($("#" + i.toString()).offset() as JQueryCoordinates).top as number;
+
+            if ( element >= documentHeight ){
+                return i;
+            } else {
+                continue
+            };
+        }
+        return toID
+}
+
+function previousPageResize(previousCursor: Array<number>): Array<number> {
+    let fromId: number = previousCursor[0] - (nextPageResize(previousCursor) - previousCursor[0]);
+    let toId: number = fromId + step;
+    return [previousCursor[0] - (nextPageResize(previousCursor) - previousCursor[0]), toId]
+}
+//#endregion
+//#endregion
+
+const step: number = 37;
+let previousCursor: Array<number>;
+let previousScale: number;
+
+window.onload = () => {     
+    previousCursor = getPageContent(0,37)
+    
+    $("#previous-page").click(() => { 
+        previousCursor = previousPageResize(previousCursor);
+        let fromId: number = (previousCursor[0] >= 0 ? previousCursor[0] : 0) as number;
+        let toId: number = (previousCursor[0] >= 0 ? previousCursor[1] : 37) as number;
+        getRecordCount()
+            .then((value) => {
+                fromId = fromId == value - 1 ? fromId - step : fromId;
+                toId = toId <= value - 1 ? toId : value - 1;
+                previousCursor = placeRecords(fromId, toId);
+            });
+        
+    });
+
+    $("#next-page").click(() => {
```

Comment:
You can do the `async` `await` thing here as well. Let me know if you need a quick intro into it. There is a lot to read on the subject but I find that it's majorly over-complicated..